ssl/quic/quic_wire.c

```diff
@@ -0,0 +1,816 @@
+        return 0;
+
+    return 1;
+}
+
+int ossl_quic_wire_decode_frame_retire_conn_id(PACKET *pkt,
+                                               uint64_t *seq_num)
+{
+    if (!expect_frame_header(pkt, OSSL_QUIC_FRAME_TYPE_RETIRE_CONN_ID)
+            || !PACKET_get_quic_vlint(pkt, seq_num))
+        return 0;
+
+    return 1;
+}
+
+int ossl_quic_wire_decode_frame_path_challenge(PACKET *pkt,
+                                               uint64_t *data)
+{
+    if (!expect_frame_header(pkt, OSSL_QUIC_FRAME_TYPE_PATH_CHALLENGE)
+            || !PACKET_get_net_8(pkt, data))
+        return 0;
+
+    return 1;
+}
+
+int ossl_quic_wire_decode_frame_path_response(PACKET *pkt,
+                                              uint64_t *data)
+{
+    if (!expect_frame_header(pkt, OSSL_QUIC_FRAME_TYPE_PATH_RESPONSE)
+            || !PACKET_get_net_8(pkt, data))
+        return 0;
+
+    return 1;
+}
+
+int ossl_quic_wire_decode_frame_conn_close(PACKET *pkt,
+                                           OSSL_QUIC_FRAME_CONN_CLOSE *f)
+{
+    uint64_t frame_type, reason_len;
+
+    /* This call matches both CONN_CLOSE_TRANSPORT and CONN_CLOSE_APP. */
+    if (!expect_frame_header_mask(pkt, OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_TRANSPORT,
+                                  1, &frame_type)
+            || !PACKET_get_quic_vlint(pkt, &f->error_code))
+        return 0;
+
+    f->is_app = ((frame_type & 1) != 0);
+
+    if (!f->is_app) {
+        if (!PACKET_get_quic_vlint(pkt, &f->frame_type))
+            return 0;
+    } else {
+        f->frame_type = 0;
+    }
+
+    if (!PACKET_get_quic_vlint(pkt, &reason_len)
+            || reason_len > SIZE_MAX)
+        return 0;
+
+    if (!PACKET_get_bytes(pkt, (const unsigned char **)&f->reason, reason_len))
+        return 0;
+
+    f->reason_len = reason_len;
+    return 1;
+}
+
+size_t ossl_quic_wire_decode_padding(PACKET *pkt)
+{
+    const unsigned char *start = PACKET_data(pkt), *end = PACKET_end(pkt),
+                        *p = start;
+
+    for (; p < end; ++p)
+        if (*p != 0)
+            break;
+
+    if (!PACKET_forward(pkt, p - start))
+        return 0;
+
+    return p - start;
+}
+
+int ossl_quic_wire_decode_frame_ping(PACKET *pkt)
+{
+    return expect_frame_header(pkt, OSSL_QUIC_FRAME_TYPE_PING);
+}
+
+int ossl_quic_wire_decode_frame_handshake_done(PACKET *pkt)
+{
+    return expect_frame_header(pkt, OSSL_QUIC_FRAME_TYPE_HANDSHAKE_DONE);
+}
+
+int ossl_quic_wire_peek_transport_param(PACKET *pkt, uint64_t *id)
+{
+    return PACKET_peek_quic_vlint(pkt, id);
+}
+
+const unsigned char *ossl_quic_wire_decode_transport_param_bytes(PACKET *pkt,
+                                                                 uint64_t *id,
+                                                                 size_t *len)
+{
+    uint64_t len_;
+    const unsigned char *b = NULL;
+
+    if (!PACKET_get_quic_vlint(pkt, id)
+            || !PACKET_get_quic_vlint(pkt, &len_))
+        return NULL;
+
+    if (len_ > SIZE_MAX
```

Comment:
Any ideas?

Reply:
Sadly, nothing comes to mind except preprocessor lines :(

Reversing the condition and excluding equality `len_ < SIZE_MAX` sort of circumvents this but is really ugly.

Reply:
I guess we shouldn't reduce the correctness of the code to satisfy compiler warnings since this is putting the cart before the horse. I'll leave this as it is for now...

If compilers start complaining about this it will definitely happen elsewhere in the codebase too, and it will probably be more productive to figure out a general strategy for it at this time.

Reply:
I'm fine with this.  I wasn't expecting a resolution -- I couldn't think of one.
Once we get warnings, we can address them.

We've had some previously BTW which we've managed to avoid.